bugsnag/bugsnag-unity
# Include more supplementary app data from WebGL

This PR makes a couple of improvements when running a Unity app on WebGL:

##### [`feat(webgl): Auto detect app version in WebGL`](https://github.com/bugsnag/bugsnag-unity/commit/45eb0bcdcf98096e32a78239fc618f920b230967)

The mobile notifiers auto-detect the app version (e.g. android from the app manifest). The version number is known in webgl and is added to the Unity metadata tab. This commit adds a LOC that only runs on webgl to pass the `Application.version` value through to the "native" JS layer.


##### [`feat(webgl): Add duration, inForeground and durationInForeground to app metadata`](https://github.com/bugsnag/bugsnag-unity/commit/9ed61eb2dad250c3f7b9d0cd800fc34ed746477d)

Unity apps deployed to mobile devices as well as webgl would inconsistenly receive data about the time that the app was running and how long it spent in the foreground, which meant that the averages were thrown off in the dashboard. This commit implements the recording of the required information when the notifier is running in a browser.

### How to test

You can build and run the example project with a target of WebGL to verify:

- `appVersion` is now supplied by default (comment out [this line](https://github.com/bugsnag/bugsnag-unity/blob/d2ddce07dd0724f504409a1a94209acbf539ccaf/example/Assets/NotifyButtonScript.cs#L47) that explicitly sets it) (look at the "app" tab for a new event in the dashboard)
- `duration`, `durationInForeground` and `inForeground` are set and look correct (try switching browser tab and letting the app run for a while before sending an error, again find this in the "app" tab of an error)
